llvm/lib/TableGen/Error.cpp

```diff
@@ -170,4 +170,11 @@ void CheckAssert(SMLoc Loc, Init *Condition, Init *Message) {
   }
 }
 
+void CheckDump(SMLoc Loc, Init *Message) {
+  if (auto *MessageInit = dyn_cast<StringInit>(Message))
+    PrintNote(MessageInit->getValue());
+  else
+    PrintNote("(dump message is not a string)");
```

Comment:
And here.

Reply:
Done - but in this case I am printing a warning instead of note. I have added a test for this.